src/Mvc/Mvc.Core/src/Formatters/SystemTextJsonInputFormatter.cs

```diff
@@ -98,5 +131,68 @@ private Stream GetInputStream(HttpContext httpContext, Encoding encoding)
 
             return new TranscodingReadStream(httpContext.Request.Body, encoding);
         }
+
+        // Keep in sync with NewtonsoftJsonInputFormatter.GetPathMetadata
+        private ModelMetadata GetPathMetadata(ModelMetadata metadata, string path)
```

Comment:
This is a fair-bit of detailed code to duplicate. Does it make sense to extra this to a separate shared file, or add as an internal API to `ModelStateDictionary` or `ModelNames`?

Reply:
`JsonPathToModelMetadataHelper` 🙈